dbms/tests/instructions/developer_instruction_en.md

```diff
@@ -0,0 +1,268 @@
+
+If you get `Connection refused` message on Mac OS X or FreeBSD, try specifying host address 127.0.0.1:
+```
+clickhouse client --host 127.0.0.1
+```
+
+You can replace production version of ClickHouse binary installed in your system with your custom built ClickHouse binary. To do that install ClickHouse on your machine following the instructions from the official website. Next, run the following:
+```
+sudo service clickhouse-server stop
+sudo cp ClickHouse/build/dbms/programs/clickhouse /usr/bin/
+sudo service clickhouse-server start
+```
+
+Note that `clickhouse-client`, `clickhouse-server` and others are symlinks to the commonly shared `clickhouse` binary.
+
+You can also run your custom built ClickHouse binary with the config file from the ClickHouse package installed on your system:
+```
+sudo service clickhouse-server stop
+sudo -u clickhouse ClickHouse/build/dbms/programs/clickhouse server --config-file /etc/clickhouse-server/config.xml
+```
+
+
+# IDE (Integrated Development Environment)
+
+If you do not know which IDE to use, we recommend that you use CLion. CLion is a commercial software, but it offers 30 day free trial period. It is also free of charge for students. CLion can be used both on Linux and on Mac OS X.
+
+KDevelop and QTCreator are another great alternatives of an IDE for developing ClickHouse. KDevelop comes in as a very handy IDE although unstable. If KDevelop crashes after a while upon opening project, you should click "Stop All" button as soon as it has opened the list of project's files. After doing so KDevelop should be fine to work with.
+
+As simple code editors you can use Sublime Text or Visual Studio Code, or Kate (all of which are available on Linux).
+
+Just in case it is worth mentioning that CLion creates by itself its own `build` path, it also selects by itself the type of build `debug` as default, for configuration it uses a version of CMake that is built into CLion and not the one installed by you, and finally CLion will use make to run build tasks instead of ninja. This is a normal behaviour, just keep that in mind to avoid confusion.
+
+
+# Writing Code
+
+The description of ClickHouse architecture can be found here: https://clickhouse.yandex/docs/en/development/architecture/
+
+The Code Style Guide: https://clickhouse.yandex/docs/en/development/style/
+
+Writing tests: https://clickhouse.yandex/docs/en/development/tests/
+
+List of tasks: https://github.com/yandex/ClickHouse/blob/master/dbms/tests/instructions/easy_tasks_sorted_en.md
+
+
+# Test Data
+
+Developing ClickHouse often requires loading realistic datasets. It is particularly important for performance testing. Especially for you we have prepared a set of anonymized data from Yandex.Metrica. It requires additionally some 3GB of free disk space. Note that this data is not required to accomplish most of development tasks.
```

Comment:
I think `Especially for you we ...` is better to change just to `We ...`

Reply:
Good point. It doesn't flow well for me neither.